nirfsg ext cal date time method name change and wait until settled parameter update (#2149)
ni/nimi-python: CHANGELOG.md

- Changed
  - Fixed the return type of `get_all_script_names` and `get_all_named_waveform_names` to remove the size parameter and return a list of strings instead of a comma-separated string
  - Changed the parameter name for waveform name from `name` to `waveform_name` in `clear_arb_waveform` and `select_arb_waveform` to be consistent with other waveform methods
  - Changed the name of below methods to be consistent with other MI drivers
    - `get_self_calibration_last_date_and_time` to `get_self_cal_last_date_and_time`
    - `get_external_calibration_last_date_and_time` to `get_ext_cal_last_date_and_time`
  - Added default value for `module` parameter in `get_self_cal_last_date_and_time` and `get_self_calibration_temperature` methods to align with documentation and allow duck-typing

>   - Updated `self_test` to not have any parameters, aligning with other MI drivers

Reviewer: ```suggestion
- Updated `self_test` to not have any parameters, aligning with other APIs in the repo
```
If we're going to take Marcos's suggestion for the other line.

  - Changed `frequency_settling` property to be of type float instead of hightime.timedelta, since it can have multiple interpretation based on `frequency_settling_units` property value

>   - Changed `interpolation_delay` and `relative_delay` properties to be of type hightime.timedelta, so that it aligns with _delay_ attributes across all drivers

Author: Fixed it.
